resources/cake/ConnextDDSExamples.cake

```diff
@@ -0,0 +1,134 @@
+/*******************************************************************************
+* (c) 2019 Copyright, Real-Time Innovations, Inc.  All rights reserved.
+* RTI grants Licensee a license to use, modify, compile, and create derivative
+* works of the Software.  Licensee has the right to distribute object form only
+* for use with RTI products.  The Software is provided "as is", with no
+* warranty of any type, including any warranty for fitness for any purpose. RTI
+* is under no obligation to maintain or support the Software.  RTI shall not be
+* liable for any incidental or consequential damages arising out of the use or
+* inability to use the software.
+******************************************************************************/
+
+var target = Argument ("target", "Default");
+var rtiArchitecture = Argument ("ConnextDDSArchitecture", "");
+var configuration = Argument("Configuration", "Release");
+
+/*
+ * Task to run Codegen and generate the the source code files and the
+ * Visual Studio Solution.
+ *
+ * This tasks expects the variables:
+ *     - NDDSHOME environment variable: set to the NDDSHOME folder
+ *     - idlName: path to the IDL file to process.
+ *     - rtiArchitecture: ConnextDDS architecture to build.
+ *
+ * All the generated files will be placed under the "src" folder. The "src"
+ * folder will be created under the current working directory.
+ */
+Task("Codegen")
+    .Does(() => {
+        string nddshomeValue = Environment.GetEnvironmentVariable("NDDSHOME");
+
+        if (string.IsNullOrEmpty(nddshomeValue)) {
+            throw new Exception("NDDSHOME environment variable was not set");
+        } else if (string.IsNullOrEmpty(rtiArchitecture)) {
+            throw new Exception("The RTI Architecture must be provided");
+        }
+
+        DirectoryPath nddshome = new DirectoryPath(nddshomeValue);
+        string rtiddsgen = "rtiddsgen";
+
+        if (IsRunningOnWindows()) {
+            rtiddsgen += ".bat";
+        }
+
+        string codegen = nddshome.Combine("bin").Combine(rtiddsgen).ToString();
```

Comment:
If I am not wrong Cake implements an implicit casting from `string` to `DirectoryPath` so I think you can simplify these path handling to:
```cs
string codegen = $"{nddshomeValue}/bin/{rtiddsgen}"
```

The same for other variables like `outputDir`, `idl`, ...
Also Cake will convert the directory path for Windows automatically like CMake does.